lib/streamlit/elements/camera_image_input.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2018-2021 Streamlit Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from streamlit.type_util import Key, to_key
+from textwrap import dedent
+from typing import Optional, cast, List
+
+import streamlit
+from streamlit.proto.CameraImageInput_pb2 import (
+    CameraImageInput as CameraImageInputProto,
+)
+
+from streamlit.report_thread import get_report_ctx
+from streamlit.state.widgets import register_widget
+from streamlit.state.session_state import (
+    WidgetArgs,
+    WidgetCallback,
+    WidgetKwargs,
+)
+
+from ..proto.Common_pb2 import (
+    FileUploaderState as FileUploaderStateProto,
+    UploadedFileInfo as UploadedFileInfoProto,
+)
+from ..uploaded_file_manager import UploadedFile, UploadedFileRec
+
+from .form import current_form_id
+from .utils import check_callback_rules, check_session_state_rules
+
+SomeUploadedSnapshotFile = Optional[UploadedFile]
+
+
+class CameraImageInputMixin:
+    def camera_image_input(
```

Comment:
personally, i think the api `camera_image_input` is too long and it should be camera but that can be decided by product. My thought process is that camera_image_input / camera have the same thought to any user so adding in a bunch of extra words to the api (_image_input) seems annoying.